src/Http/Routing/src/Builder/RouteHandlerFilterExtensions.cs

```diff
@@ -29,9 +32,15 @@ public static RouteHandlerBuilder AddFilter(this RouteHandlerBuilder builder, IR
     /// <typeparam name="TFilterType">The type of the <see cref="IRouteHandlerFilter"/> to register.</typeparam>
     /// <param name="builder">The <see cref="RouteHandlerBuilder"/>.</param>
     /// <returns>A <see cref="RouteHandlerBuilder"/> that can be used to further customize the route handler.</returns>
-    public static RouteHandlerBuilder AddFilter<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] TFilterType>(this RouteHandlerBuilder builder) where TFilterType : IRouteHandlerFilter, new()
+    public static RouteHandlerBuilder AddFilter<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] TFilterType>(this RouteHandlerBuilder builder) where TFilterType : IRouteHandlerFilter
     {
-        builder.RouteHandlerFilters.Add(new TFilterType());
+        builder.RouteHandlerFilterFactories.Add((methodInfo, next) => async (context) =>
+        {
+            var type = typeof(TFilterType);
+            var filterFactory = ActivatorUtilities.CreateFactory(type, Array.Empty<Type>());
```

Comment:
The factory should be resolved once and used to create the factory the filter per request.